### PR TITLE
U4-4700 List View Doesn't Respect User Permissions

### PR DESCRIPTION
**Pull request by warrenbuckley**
This creates a simpler easier to read JSON object to be used in the listview to disable certain UI buttons, such as copy, move, publish, unpublish, delete & create based on the users permission for browsing the current node with the ListView property editor in it.

As the permissions are some ugly array of strings such as `C, U, Z` etc so we map to a neater model of `canEdit, canMove` etc

With a little help from @zpqrtbnk was able to debug & understand why not all permissions were being pulled through as needed & why when updating them the cache for this was not invalidated.

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/listview.controller.js [hunk 1]
```diff
@@ -59,6 +59,37 @@ function listViewController($rootScope, $scope, $routeParams, $injector, $cookie
...
+        //Get Current User to check if their usertype is an admin (overwrites all permissions!)
+        userService.getCurrentUser().then(function(data){
+            
+            //Fetch current node allowed actions for the current user
+            //This is the current node & not each individual child node in the list
+            var currentUserPermissions = editorState.current.allowedActions;
+                
+            //Create a nicer model rather than the funky & hard to remember permissions strings
+            $scope.currentNodePermissions = {
+                "isAdminUser": data.userType.toLowerCase() === "admin",
```
Comment (Shazwazza) [inline src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/listview.controller.js]: We shouldn't really do this, it doesn't matter if the user is an 'admin' user type, a user type is simply a way to flag default permissions, their permissions could be changed by another admin

**Comment by Shazwazza** [comment]
So the way this is working in this PR is that we're making an assumption that all child node permissions are the same as the current node's permissions which may not be true. 

What we need to do instead is:

* Create a new endpoint to check a content item's permissions.
* This endpoint should accept an array of Ids in a POST to return the permissions for a range of nodes
* Whenever the admin clicks on a child node in the list view, we check all common permissions for all nodes currently selected and filter the buttons accordingly
* If an admin clicks on 'select all' items, we can send a single request to check permissions for all of those nodes

I'm happy to do this but would like feedback on this since you guys have discussed some of this already

**Comment by nul800sebastiaan** [comment]
> we're making an assumption that all child node permissions are the same as the current node's permissions
Haven't looked at this yet, but we discussed that it could be possible some child nodes had different permissions, so that wasn't the intention.

**Comment by Shazwazza** [comment]
Cool, I'm updating this now to do what I mentioned in my notes above.

**Comment by nul800sebastiaan** [comment]
:+1:

**Comment by warrenbuckley** [comment]
Yes @Shazwazza I think we discussed that yes child nodes may not have the same permissions, but I was not sure at the time in terms of UI/UX

Say you select 5 nodes, 4 of them you are allowed to delete but the 5th you've selected does not. Do you hide/disable the button now or do you enable it still & give a message/warning that the 5th item could not be deleted due to permissions.

So I can't remember why it was fully agreed to just check the current nodes permission & assume the same would be applied to child nodes or maybe I just done it (barely remember yesterday let alone almost 2 weeks ago)

Seems like you have a battle plan for this @Shazwazza :+1:

**Comment by nul800sebastiaan** [comment]
Next time we'll just have to write these things down. It was my understanding that when you select one or more items that don't have correct permissions the action should be impossible.

File: src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/listview.controller.js (hunk 1 above)
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/views/propertyeditors/listview/listview.controller.js]: Yeh no problem, again it was just an assumption made by me at the time.

**Pull request merged**